### PR TITLE
tECDSA signing: Setup of the test loop and signing executor

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Refs: https://github.com/keep-network/keep-core/issues/3042
Depends on: https://github.com/keep-network/keep-core/pull/3209

Here we are setting up the tECDSA signing test loop and the tECDSA signing protocol executor, without executing the actual signing logic yet. Specific changes are summarized below.

### Test loop setup

The signing loop was set up using a temporary chain mock that generates a new `tbtc.SignatureRequestedEvent` periodically targeting the recently created wallet. The signature request is about signing the hash of the block the event was generated at. The `tbtc` application node listens for the aforementioned events and decides about joining the signing process if the node controls at least one of the requested wallet's signers. 

Worth mentioning that this trigger of the signing process is temporary, and ultimately, the signing process will not be initiated explicitly by the chain. Instead of that, the `tbtc` node will observe domain events emitted by the `Bridge` contract` and react to them by constructing and signing specific BTC transactions or other data required by the given use case.

### Signing protocol executor setup

Once the `tbtc` node decides to join the given signing execution, it must set up the tECDSA signing protocol executor properly. This includes setting up the right broadcast channel and determining all the input parameters required by the tECDSA signing executor. This PR addresses those requirements within the `joinSigningIfEligible` function.

Additionally, we are also exposing the executor function itself.  The function is called `pkg/tecdsa/signing.Execute` and this PR focuses on defining its public API. The actual logic is just a stub at the moment and will be addressed in further PRs.

### Next steps

This PR is just a part of the entire work regarding the tECDSA signing test loop. Please refer to the description of https://github.com/keep-network/keep-core/issues/3042 for the full roadmap and the current status of the work.

### CODE REVIEW

**Comment by lukasz-zimnoch** [comment]
The code is ready for review. I'll undraft once https://github.com/keep-network/keep-core/pull/3209 and manual testing of this PR is done.

**Review by tomaszslabon — APPROVED** [review]

File: pkg/tbtc/node.go [hunk 1]
```diff
@@ -345,3 +347,158 @@ func (n *node) joinDKGIfEligible(seed *big.Int, startBlockNumber uint64) {
...
+	logger.Infof(
+		"checking eligibility for signature of message [%v]",
+		message,
+	)
+
+	if signers := n.walletRegistry.getSigners(
+		walletPublicKey,
+	); len(signers) > 0 {
+		logger.Infof(
+			"joining signature of message [%v] and "+
```
Comment (pdyraga) [inline pkg/tbtc/node.go:373]:
```suggestion
			"joining signature of message [%v] "+
```

File: pkg/tbtc/node.go [hunk 2]
```diff
@@ -345,3 +347,158 @@ func (n *node) joinDKGIfEligible(seed *big.Int, startBlockNumber uint64) {
...
+		// All signers belong to one wallet. Take that wallet from the
+		// first signer.
+		wallet := signers[0].wallet
+		// Actual wallet signing group size may be different from the
+		// `GroupSize` parameter of the chain config.
+		signingGroupSize := len(wallet.signingGroupOperators)
+		// The dishonest threshold for the wallet signing group must be
+		// also calculated using the actual wallet signing group size.
+		signingGroupDishonestThreshold := signingGroupSize -
+			n.chain.GetConfig().HonestThreshold
```
Comment (pdyraga) [inline pkg/tbtc/node.go:388]: This piece of code feels like something that should be extracted out of `node.go` into some more specialized component.

**Review by pdyraga — APPROVED** [review]

File: pkg/tbtc/node.go (hunk 2 above)
Comment (lukasz-zimnoch) [reply pkg/tbtc/node.go:388]: I will leave that for now and think about it at the end, once retries and other auxiliary signing components are implemented.

File: pkg/chain/ethereum/tbtc.go [hunk 3]
```diff
@@ -482,12 +490,57 @@ func (mwr *mockWalletRegistry) SubmitDKGResult(
 		}(handler)
 	}
 
-	mwr.activeGroup = groupPublicKeyBytes
-	mwr.activeGroupOperableBlock = new(big.Int).Add(
+	mwr.activeWallet = groupPublicKeyBytes
+	mwr.activeWalletOperableBlock = new(big.Int).Add(
 		mwr.currentDkgStartBlock,
-		big.NewInt(150),
+		big.NewInt(200),
```
[the remark this answers is not in the document]
Comment (lukasz-zimnoch) [reply pkg/chain/ethereum/tbtc.go:510]: Addressed that in https://github.com/keep-network/keep-core/pull/3228/commits/16e21345f6cbb1d87275463d4fe9939987d63321

File: pkg/tbtc/node.go (hunk 1 above)
Comment (lukasz-zimnoch) [reply pkg/tbtc/node.go:373]: Addressed that in https://github.com/keep-network/keep-core/pull/3228/commits/10e6f8ca69e15022c1501ca04d5253b037645d06

**Pull request merged**